File: rb/1.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  hash = {}

  nums.each_with_index do |num, i|
    if hash.key? (target - num)
      break [hash[target - num], i]
    else
      hash[num] = i
    end
  end
end

File: leetcode/py/1.py
class Solution:
    def twoSum(self, nums, target):
        map = {}

        for idx, n in enumerate(nums):
            if n in map:
                return [map[n], idx]

            map[target-n] = idx

class Solution:
    def twoSum(self, nums, target):
        num_idx_map = {}

        for i, num in enumerate(nums):
            if target - num in num_idx_map:
                return [num_idx_map[target - num], i]
            num_idx_map[num] = i

